pkg/rules/handlers.go

```diff
@@ -0,0 +1,243 @@
+package rules
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/http/types"
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+
+	"github.com/go-chi/chi/v5"
+	oathkeeper "github.com/ory/oathkeeper-client-go"
+)
+
+type API struct {
+	service ServiceInterface
+
+	logger logging.LoggerInterface
+}
+
+func (a *API) RegisterEndpoints(mux *chi.Mux) {
+	mux.Get("/api/v0/rules", a.handleList)
+	mux.Get("/api/v0/rules/{id}", a.handleDetail)
+	mux.Post("/api/v0/rules", a.handleCreate)
+	mux.Put("/api/v0/rules/{id}", a.handleUpdate)
+	mux.Delete("/api/v0/rules/{id}", a.handleRemove)
+}
+
+func (a *API) handleList(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	pagination := types.ParsePagination(r.URL.Query())
+	limit := pagination.Size
+	offset := (pagination.Page - 1) * pagination.Size
```

Comment:
point of the service is to abstract and deal with parameters in its own way, so i d pass the pagination stuff as is and move this chunk in the service layer

https://github.com/canonical/identity-platform-admin-ui/pull/157/files#diff-f527fb4a79ca6df25497874fd4ddf6c1c996e72bb901367dacca55664e5ee2b7R37

Reply:
done